packages/lit-dev-content/src/github/github-signin.ts

```diff
@@ -65,14 +74,32 @@ export const signInToGithub = async (
   window.removeEventListener('beforeunload', closePopupIfParentCloses);
   if (codeOrClosed === 'closed') {
     pendingCode.abort();
-    // TODO(aomarks) Display a more helpful message to the user in the UI.
-    throw new Error('Popup was closed too early!');
+    throw new LitDevError({
+      heading: 'GitHub window closed',
+      message:
+        'It looks like you closed the GitHub window. If you still' +
+        " want to create a GitHub gist, you'll need to click " +
+        '"Authorize lit" when prompted by the popup window.',
+    });
   }
   popup.close();
   const code = codeOrClosed;
   if (code.error !== undefined) {
-    // TODO(aomarks) Display a more helpful message to the user in the UI.
-    throw new Error(`Error getting code: ${code.error}`);
+    if (code.error === 'access_denied') {
+      throw new LitDevError({
```

Comment:
Nit: Do we need to open the error modal if the user decides to opt out of the authorization?